### PR TITLE
Add check-for-av-emulation-using-virtualallocexnuma.yml

### PR DESCRIPTION
**Pull request by jtothej**
Add check-for-av-emulation-using-virtualallocexnuma.yml

### CODE REVIEW

File: anti-analysis/anti-av/check-for-av-emulation-using-virtualallocexnuma.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    mbc:
+      - Anti-Behavioral Analysis::Emulator Evasion [B0005]
+    references:
+      - https://www.purpl3f0xsecur1ty.tech/2021/03/30/av_evasion.html
+    examples:
+      - 17b31898c3613d6e86d06260c7e6f7738d3afdf0f0a4010e2f5259c502d38b09:0x10001A1D
+  features:
+    - and:
+      - api: kernel32.VirtualAllocExNuma
+      - api: kernel32.GetCurrentProcess
```
Comment (mr-tz) [inline anti-analysis/anti-av/check-for-av-emulation-using-virtualallocexnuma.yml:17]: I don't think just these two calls indicate any anti-AV. Can you elaborate?

Comment (jtothej) [reply anti-analysis/anti-av/check-for-av-emulation-using-virtualallocexnuma.yml:17]: I have not tested it by myself but multiple sources claim that `VirtualAllocExNuma` is not/was not handled well by AV emulation engines and thus might be used for emulation/sandbox detection. In all instances (sample source code and malware) I found this is typically done by calling `VirtualAllocExNuma` with current process handle and checking if return value is NULL.

Comment (mr-tz) [reply anti-analysis/anti-av/check-for-av-emulation-using-virtualallocexnuma.yml:17]: Benign code could work the same though, see ,e.g., https://learn.microsoft.com/en-us/windows/win32/memory/allocating-memory-from-a-numa-node. So, I don't think that's a check we could rely on here. Thoughts?

Comment (jtothej) [reply anti-analysis/anti-av/check-for-av-emulation-using-virtualallocexnuma.yml:17]: Absolutely agree it is completely benign - at the same time it seemed to be quite unique/rare combination (VT: ~35k samples importing both, ~1.5k samples with 0 detections). I can now definitely see samples using both functions in intended way so let's reject this PR. Thank you for the review!

Comment (williballenthin) [reply anti-analysis/anti-av/check-for-av-emulation-using-virtualallocexnuma.yml:17]: could we perhaps rename the rule to simply describe the behavior for what it is, and then let some downstream system use it for detection, if desired?

the distribution on VT is interesting so i think there's value to having this rule. a capa rule doesn't have to describe only malicious things.

"allocate memory on this NUMA node" or similar?

Comment (mr-tz) [reply anti-analysis/anti-av/check-for-av-emulation-using-virtualallocexnuma.yml:17]: Good point, we currently (only) include `VirtualAllocExNuma` in the `allocate memory` rule.

**Pull request closed without merging**